### PR TITLE
move Windows code into a module

### PR DESCRIPTION
**Pull request by Be-ing**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -69,23 +69,10 @@ use std::thread::{self, JoinHandle};
...
-#[cfg(windows)]
-mod vs_instances;
-#[cfg(windows)]
-mod windows_sys;
-
-pub mod windows_registry;
+mod windows;
+// Regardless of whether this should be in this crate's public API,
+// it has been since 2015, so don't break it.
+pub use windows::find_tools as windows_registry;
```
Comment (Be-ing) [inline src/lib.rs:75]: Maybe make this `#[doc(hidden)]`?

File: src/lib.rs [hunk 2]
```diff
@@ -2554,7 +2541,7 @@ impl Build {
             traditional
         };
 
-        let cl_exe = windows_registry::find_tool(target, "cl.exe");
+        let cl_exe = windows::find_tools::find_tool(target, "cl.exe");
```
Comment (NobodyXu) [inline src/lib.rs:2544]: there's no need to change this line:

```suggestion
        let cl_exe = windows_registry::find_tool(target, "cl.exe");
```

File: src/lib.rs [hunk 3]
```diff
@@ -3025,7 +3012,7 @@ impl Build {
 
                     if lib.is_empty() {
                         name = String::from("lib.exe");
-                        let mut cmd = match windows_registry::find(&target, "lib.exe") {
+                        let mut cmd = match windows::find_tools::find(&target, "lib.exe") {
```
Comment (NobodyXu) [inline src/lib.rs:3015]: Same here:

```suggestion
                        let mut cmd = match windows_registry::tools::find(&target, "lib.exe") {
```

**Review by NobodyXu — CHANGES_REQUESTED** [review]
Thanks, overall it LGTM but has two minor suggestions.

Also, can you please run gen-windows-sys-binding locally to make sure it works with the changed location?

File: src/lib.rs (hunk 2 above)
Comment (Be-ing) [reply src/lib.rs]: done

File: src/lib.rs (hunk 3 above)
Comment (Be-ing) [reply src/lib.rs]: done

**Comment by Be-ing** [comment]
> Also, can you please run gen-windows-sys-binding locally to make sure it works with the changed location?

Ah yeah, it needed a little adjustment since moving gen-windows-sys-binding to the dev-tools directory in #927. Now it works regardless of how you run gen-windows-sys-binding.

**Comment by Be-ing** [comment]
This is easier to navigate now:

```
cc-rs on  windows_mod [$] is 📦 v1.0.84 via 🦀 v1.75.0 on ☁  (auto) 
❯ tree
.
├── Cargo.toml
├── dev-tools
│   ├── cc-test
│   │   ├── build.rs
│   │   ├── Cargo.toml
│   │   ├── src
│   │   │   ├── aarch64.asm
│   │   │   ├── aarch64.S
│   │   │   ├── armv7.S
│   │   │   ├── bar1.c
│   │   │   ├── bar2.c
│   │   │   ├── baz.cpp
│   │   │   ├── compile_error.c
│   │   │   ├── cuda.cu
│   │   │   ├── dummy.c
│   │   │   ├── expand.c
│   │   │   ├── foo.c
│   │   │   ├── i686.asm
│   │   │   ├── i686.S
│   │   │   ├── include
│   │   │   │   └── foo.h
│   │   │   ├── lib.rs
│   │   │   ├── msvc.c
│   │   │   ├── NMakefile
│   │   │   ├── opt_linkage.c
│   │   │   ├── riscv64gc.S
│   │   │   ├── windows.c
│   │   │   ├── x86_64.asm
│   │   │   └── x86_64.S
│   │   └── tests
│   │       ├── all.rs
│   │       └── output.rs
│   └── gen-windows-sys-binding
│       ├── Cargo.toml
│       ├── src
│       │   └── main.rs
│       └── windows_sys.list
├── LICENSE-APACHE
├── LICENSE-MIT
├── README.md
├── src
│   ├── bin
│   │   └── gcc-shim.rs
│   ├── command_helpers.rs
│   ├── lib.rs
│   ├── os_pipe
│   │   ├── mod.rs
│   │   ├── unix.rs
│   │   └── windows.rs
│   ├── parallel
│   │   ├── async_executor.rs
│   │   ├── job_token
│   │   │   ├── mod.rs
│   │   │   ├── unix.rs
│   │   │   └── windows.rs
│   │   └── mod.rs
│   ├── tool.rs
│   └── windows
│       ├── com.rs
│       ├── find_tools.rs
│       ├── mod.rs
│       ├── registry.rs
│       ├── setup_config.rs
│       ├── vs_instances.rs
│       ├── winapi.rs
│       └── windows_sys.rs
└── tests
    ├── cc_env.rs
    ├── cflags.rs
    ├── cxxflags.rs
    ├── support
    │   └── mod.rs
    └── test.rs

16 directories, 58 files
```

**Review by NobodyXu — APPROVED** [review]
Thanks!

**Pull request merged**